feature: add overlayOptions from the origostyle-object

jokd opened the pull request:
Fixes #1760
When setting origostyle.overlayOptions.offset for a feature it will offset the popup overlay.
Use with caution because when having multiple features in the carousel it will only check the property of the first feature.

Eg:
![image](https://user-images.githubusercontent.com/6848075/235641079-d2c16ddc-bf8f-45fd-aee0-7102907683e7.png)

Can be used like this:
```
const viewer = origo.api();

const layerOptions = {
group: 'root',
name: 'markerLayer',
title: 'Markers',
zIndex: 6,
styleByAttribute: true,
queryable: true,
removable: true,
visible: true,
source: 'none',
type: 'FEATURE',
attributes: [
{
html: '{{content}}'
}
],
style: viewer.getStylewindow().getStyleFunction
};

const layer = viewer.addLayer(layerOptions);
const coords = [1575114, 8396023];
const newFeature = viewer.getMapUtils().createPointFeature(coords, null);
newFeature.set('origostyle', {pointType: 'marker', overlayOptions: { offset:[0, -35] }});
newFeature.set('content', 'This markers content!');
layer.getSource().addFeature(newFeature);
```

johnnyblasta commented:
Tested and works with the API, but wouldn't it be cool to be able to set it in config on the style also?

jokd commented:
> Tested and works with the API, but wouldn't it be cool to be able to set it in config on the style also?

Added possibility to make this in the style object as well:
```
    "origo-logo": [
      [
        {
          "overlayOptions": {
            "offset":[15, 15],
            "autoPan": false,
            "positioning": "center-right"
          },
          "label": "Origokommuner",
.....

All Openlayers overlay options (https://openlayers.org/en/latest/apidoc/module-ol_Overlay-Overlay.html) are available however probably only positioning, autoPan and offset are useful.
Added styling to support bottom-center, center-left, top-center and center-right positioning
```

johnnyblasta reviewed (approved):
LGTM
